### PR TITLE
perf: speedup sentry shell initialization by ~30%

### PR DESCRIPTION
**Pull request by joshuarli**
I was working with `devservices` a lot recently, and sentry cli startup time was getting pretty annoying. `sentry devservices attach foo` which basically noops but goes through the entire initialization takes around 4.5 - 5 seconds on my machine. I sped it up to 3 - 3.5 seconds.

I generated a cProfile of `initialize_app`, a large portion of it was spent in regex CPU at module-level in this one module. Lazily populating it is the easiest fix, I'm hoping all tests pass.

Before:

before

After:

after

### CODE REVIEW

**Comment by joshuarli** [comment]
Some other interesting things worth looking at:

- `assert_package_not_installed` cumtime is roughly 0.4s, though pkg_resources is notoriously slow to both import and use...
  - A fast fix could just be to only do that in CI.
- our `import_submodules` costs ~0.7s cumtime, maybe something smarter could be done with it
- ~0.4s more regex cumtime by `message_filters.py` module-level, which is descended from a long chain of inits, this can probably be made lazy too

**Comment by joshuarli** [comment]
Confirmed removing `assert_package_not_installed` saves ~0.4s. Wall clock time dropped by that amount consistently. Lemme try the other stuff, deferring deciding whether or not to split it up into separate PRs.

Of course, assuming that it isn't useful to us anymore. @scefali any comments?

**Comment by joshuarli** [comment]
By profiling `sentry.runner.settings::configure()` I get nearly the complete picture, this time with register_plugins not nooping - cumtime ~1s. So another potential optimization candidate.

File: src/sentry_plugins/heroku/__init__.py [hunk 1]
```diff
@@ -1,5 +1 @@
 from __future__ import absolute_import
-
-from sentry_plugins.base import assert_package_not_installed
-
-assert_package_not_installed("sentry-heroku")
```
Comment (scefali) [inline src/sentry_plugins/heroku/__init__.py:5]: @joshuarli Removing these is totally fine

Comment (joshuarli) [reply src/sentry_plugins/heroku/__init__.py:5]: Perfect, thanks.

**Comment by joshuarli** [comment]
Okay, here's something more complete (love seeing the CPU cache warm up lol):

```
joshua.li@tim-apple sentry master $ time sentry shell -c exit
sentry shell -c exit  4.26s user 0.74s system 98% cpu 5.094 total
joshua.li@tim-apple sentry master $ time sentry shell -c exit
sentry shell -c exit  3.69s user 0.60s system 98% cpu 4.357 total
joshua.li@tim-apple sentry master $ time sentry shell -c exit
sentry shell -c exit  3.73s user 0.62s system 98% cpu 4.435 total
joshua.li@tim-apple sentry master $ time sentry shell -c exit
sentry shell -c exit  3.64s user 0.60s system 98% cpu 4.297 total

joshua.li@tim-apple sentry master $ git c -
Switched to branch 'perf/speedup-sentry-startup-part-1'
Your branch is up to date with 'origin/perf/speedup-sentry-startup-part-1'.

joshua.li@tim-apple sentry perf/speedup-sentry-startup-part-1 $ time sentry shell -c exit
sentry shell -c exit  2.38s user 0.53s system 96% cpu 3.010 total
joshua.li@tim-apple sentry perf/speedup-sentry-startup-part-1 $ time sentry shell -c exit
sentry shell -c exit  2.37s user 0.53s system 98% cpu 2.959 total
joshua.li@tim-apple sentry perf/speedup-sentry-startup-part-1 $ time sentry shell -c exit
sentry shell -c exit  2.42s user 0.55s system 96% cpu 3.058 total
joshua.li@tim-apple sentry perf/speedup-sentry-startup-part-1 $ time sentry shell -c exit
sentry shell -c exit  2.41s user 0.54s system 97% cpu 3.034 total
```

So, safe to say, `sentry shell` startup is ~30% faster overall.

File: src/sentry_plugins/heroku/__init__.py (hunk 1 above)
Comment (joshuarli) [reply src/sentry_plugins/heroku/__init__.py:5]: Decoupled into https://github.com/getsentry/sentry/pull/18628.

**Review by kamilogorek — APPROVED** [review]
LGTM. We definitely don't need these translations right away.

**Pull request merged**